app/src/main/java/org/schabi/newpipe/player/ui/VideoPlayerUi.java

```diff
@@ -1219,6 +1247,53 @@ public boolean onMenuItemClick(@NonNull final MenuItem menuItem) {
         return false;
     }
 
+    private void onQualityItemClick(@NonNull final MenuItem menuItem) {
+        final int menuItemIndex = menuItem.getItemId();
+        @Nullable final MediaItemTag currentMetadata = player.getCurrentMetadata();
+        if (currentMetadata == null || currentMetadata.getMaybeQuality().isEmpty()) {
+            return;
+        }
+
+        final MediaItemTag.Quality quality = currentMetadata.getMaybeQuality().get();
+        final List<VideoStream> availableStreams = quality.getSortedVideoStreams();
+        final int selectedStreamIndex = quality.getSelectedVideoStreamIndex();
+        if (selectedStreamIndex == menuItemIndex || availableStreams.size() <= menuItemIndex) {
+            return;
+        }
+
+        player.saveStreamProgressState();
+        final String newResolution = availableStreams.get(menuItemIndex).getResolution();
+        player.setRecovery();
+        player.setPlaybackQuality(newResolution);
+        player.reloadPlayQueueManager();
+
+        binding.qualityTextView.setText(menuItem.getTitle());
+    }
+
+    private void onAudioTrackItemClick(@NonNull final MenuItem menuItem) {
+        final int menuItemIndex = menuItem.getItemId();
+        @Nullable final MediaItemTag currentMetadata = player.getCurrentMetadata();
+        if (currentMetadata == null || currentMetadata.getMaybeAudioTrack().isEmpty()) {
+            return;
+        }
+
+        final MediaItemTag.AudioTrack audioTrack =
+                currentMetadata.getMaybeAudioTrack().get();
+        final List<AudioStream> availableStreams = audioTrack.getAudioStreams();
+        final int selectedStreamIndex = audioTrack.getSelectedAudioStreamIndex();
+        if (selectedStreamIndex == menuItemIndex || availableStreams.size() <= menuItemIndex) {
+            return;
+        }
+
+        player.saveStreamProgressState();
+        final String newAudioTrack = availableStreams.get(menuItemIndex).getAudioTrackId();
+        player.setRecovery();
+        player.setAudioTrack(newAudioTrack);
+        player.reloadPlayQueueManager();
```

Comment:
This part is also present in PlayQueueActivity. It's better to create a function on the player that combines these actions (maybe directly inside `setAudioTrack`, unless it's used in other places)